onnx/backend/test/cpp/driver_test.cc

```diff
@@ -0,0 +1,249 @@
+            &model_,
+            weightCount,
+            &weightDescriptors,
+            &graph),
+        ONNXIFI_STATUS_SUCCESS);
+
+    for (const auto& proto_test_data : protos_) {
+      std::vector<onnxTensorDescriptorV1> input_descriptor, output_descriptor,
+          result_descriptor;
+      for (const auto& input : proto_test_data.inputs_) {
+        input_descriptor.push_back(
+            ONNX_NAMESPACE::testing::ProtoToOnnxTensorDescriptor(input));
+      }
+      int output_count = 0;
+      for (auto& output : proto_test_data.outputs_) {
+        output_count++;
+        output_descriptor.push_back(
+            ONNX_NAMESPACE::testing::ProtoToOnnxTensorDescriptor(output));
+        onnxTensorDescriptorV1 result;
+        result.tag = ONNXIFI_TAG_TENSOR_DESCRIPTOR_V1;
+        std::string name_string =
+            "output_" + ONNX_NAMESPACE::to_string(output_count);
+        result.name = name_string.c_str();
+        result.dataType = output.data_type();
+        result.memoryType = ONNXIFI_MEMORY_TYPE_CPU;
+        std::vector<uint64_t> shape_values(
+            output.dims().begin(), output.dims().end());
+        result.dimensions = shape_values.size();
+        result.shape = shape_values.data();
+        std::vector<uint8_t> raw_data(output.raw_data().size(), 0);
+        result.buffer = (onnxPointer)raw_data.data();
+        result_descriptor.emplace_back(std::move(result));
+      }
+      EXPECT_EQ(
+          lib.onnxSetGraphIO(
+              graph,
+              input_descriptor.size(),
+              input_descriptor.data(),
+              result_descriptor.size(),
+              result_descriptor.data()),
+          ONNXIFI_STATUS_SUCCESS);
+
+      onnxMemoryFenceV1 inputFence, outputFence;
+      inputFence.tag = ONNXIFI_TAG_MEMORY_FENCE_V1;
+      outputFence.tag = ONNXIFI_TAG_MEMORY_FENCE_V1;
+      inputFence.type = ONNXIFI_SYNCHRONIZATION_EVENT;
+      outputFence.type = ONNXIFI_SYNCHRONIZATION_EVENT;
+
+      EXPECT_EQ(
+          lib.onnxRunGraph(graph, &inputFence, &outputFence),
+          ONNXIFI_STATUS_SUCCESS);
+      EXPECT_EQ(lib.onnxWaitEvent(outputFence.event), ONNXIFI_STATUS_SUCCESS);
+      for (int i = 0; i < output_descriptor.size(); i++) {
+        auto output_size = GetDescriptorSize(&output_descriptor[i]);
+        for (int j = 0; j < output_size; j++) {
+          EXPECT_EQ(
+              IsDescriptorEqual(output_descriptor[i], result_descriptor[i]),
+              true);
+          ;
+        }
+      }
+    }
+    EXPECT_EQ(lib.onnxReleaseGraph(graph), ONNXIFI_STATUS_SUCCESS);
+#endif
+  }
+};
+
+/**
+ *	When testing backend, we do not specify target backend here,
+ *	but always use the by-default backend of onnxifi_loader.
+ *	Check onnxifi_loader.h for more detail of how to indicate specific
+ *backend.
+ */
+TEST_P(ONNXCppDriverTest, ONNXCppDriverUnitTest){
+	onnxifi_library lib;
+	onnxBackendID backendID;
+	onnxBackend backend;
+#if !ONNXIFI_DUMMY_BACKEND
```

Comment:
I think you could use regular `if` here (not #if)

Reply:
Sounds decent